pkg/scheduler/scheduler_test.go

```diff
@@ -18,23 +18,1232 @@ package scheduler
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              "deleting-binding",
+					DeletionTimestamp: &metav1.Time{Time: time.Now()},
+				},
+			},
+			expectSchedule: false,
+			expectError:    false,
+		},
+		{
+			name: "binding with replicas changed",
+			binding: &workv1alpha2.ClusterResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "replicas-changed-binding",
+					Annotations: map[string]string{
+						util.PolicyPlacementAnnotation: `{"replicaScheduling":{"replicaDivisionPreference":"Weighted","replicaSchedulingType":"Divided"}}`,
+					},
+				},
+				Spec: workv1alpha2.ResourceBindingSpec{
+					Replicas: 2,
+					Placement: &policyv1alpha1.Placement{
+						ReplicaScheduling: &policyv1alpha1.ReplicaSchedulingStrategy{
+							ReplicaDivisionPreference: policyv1alpha1.ReplicaDivisionPreferenceWeighted,
+							ReplicaSchedulingType:     policyv1alpha1.ReplicaSchedulingTypeDivided,
+						},
+					},
+				},
+				Status: workv1alpha2.ResourceBindingStatus{
+					SchedulerObservedGeneration: 1,
+				},
+			},
+			scheduleFunc: func(_ context.Context, _ *workv1alpha2.ResourceBindingSpec, _ *workv1alpha2.ResourceBindingStatus, _ *core.ScheduleAlgorithmOption) (core.ScheduleResult, error) {
+				return core.ScheduleResult{
+					SuggestedClusters: []workv1alpha2.TargetCluster{
+						{Name: "cluster1", Replicas: 1},
+						{Name: "cluster2", Replicas: 1},
+					},
+				}, nil
+			},
+			expectSchedule: true,
+			expectError:    false,
+			expectedPatches: []string{
+				`{"metadata":{"annotations":{"policy.karmada.io/applied-placement":"{\"replicaScheduling\":{\"replicaSchedulingType\":\"Divided\",\"replicaDivisionPreference\":\"Weighted\"}}"}},"spec":{"clusters":[{"name":"cluster1","replicas":1},{"name":"cluster2","replicas":1}]}}`,
+				`{"status":{"conditions":[{"message":"Binding has been scheduled successfully.","reason":"Success","status":"True","type":"Scheduled"}],"schedulerObservedGeneration":null}}`,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClient := karmadafake.NewSimpleClientset(tt.binding)
+			fakeRecorder := record.NewFakeRecorder(10)
+			mockAlgo := &mockAlgorithm{
+				scheduleFunc: tt.scheduleFunc,
+			}
+
+			s := &Scheduler{
+				KarmadaClient:        fakeClient,
+				clusterBindingLister: &fakeClusterBindingLister{binding: tt.binding},
+				eventRecorder:        fakeRecorder,
+				Algorithm:            mockAlgo,
+			}
+
+			err := s.doScheduleClusterBinding(tt.binding.Name)
+
+			if tt.expectError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+
+			actions := fakeClient.Actions()
+			patchActions := filterPatchActions(actions)
+
+			if tt.expectSchedule {
+				assert.Len(t, patchActions, len(tt.expectedPatches), "Expected %d patch actions, got %d", len(tt.expectedPatches), len(patchActions))
+				for i, expectedPatch := range tt.expectedPatches {
+					if i < len(patchActions) {
```

Comment:
Why add this judgment?

In addition, the judgment logic of the TestDoScheduleClusterBinding and TestDoScheduleBinding functions is different. Can the TestDoScheduleClusterBinding and TestDoScheduleBinding functions directly determine the binding field obtained at the same time?

Reply:
I'll remove those judgements and refractor both tests to be consistent.